src/WinRT.Runtime2/InteropServices/Activation/WindowsRuntimeActivationFactory.cs

```diff
@@ -79,6 +84,10 @@ public static WindowsRuntimeObjectReference GetActivationFactory(string runtimeC
 
     /// <param name="iid">The IID of the interface pointer (from the resolved activation factory) to wrap in the returned object reference.</param>
     /// <inheritdoc cref="GetActivationFactory(string)"/>
+    [Obsolete(WindowsRuntimeConstants.PrivateImplementationDetailObsoleteMessage,
+        DiagnosticId = WindowsRuntimeConstants.PrivateImplementationDetailObsoleteDiagnosticId,
+        UrlFormat = WindowsRuntimeConstants.CsWinRTDiagnosticsUrlFormat)]
+    [EditorBrowsable(EditorBrowsableState.Never)]
     public static WindowsRuntimeObjectReference GetActivationFactory(string runtimeClassName, in Guid iid)
```

Comment:
We will probably need a solution for interop interfaces that someone may want to QI for and use, but we can address that later with a higher level wrapper.

Reply:
Yeah for now it's usable already, but callers will need to get the pointer and then marshal with `ComWrappers` on their end. We can definitely think of some higher level API where we'd handle that part ourselves. Like idk, `GetActivationFactoryInterface<T>` or something, we can do that later, agreed 🙂